src/Microsoft.Azure.WebJobs.Logging/Internal/TimeBucket.cs

```diff
@@ -27,5 +29,63 @@ public static long ConvertToBucket(DateTime dt)
             }
             return min;
         }
+
+        // Used to compute epoch. 
+        const long ticksBaseline = 633663648000000000L;
+        const long ticksPerMonth = 26785144663987L;
+        
+        public static DateTime CommonEpoch = new DateTime(0, DateTimeKind.Utc); // maps to epoch 0 
+        public static string CommonEpochSuffix = "common";
+
+        public static long GetEpochNumberFromTable(CloudTable table)
+        {
+            string tableName = table.Name;
+            if (tableName.Length < 5)
+            {
+                return -1;
+            }
+            string suffix = tableName.Substring(tableName.Length - 5, 5);
+            long epoch;
+            if (long.TryParse(suffix, out epoch))
+            {
+                return epoch;
+            }
+            return -1;
+        }
+
+        // Epoch must be positive, orderd integers.
+        // Use YYYYMM 
+        private static long GetEpochNumber(DateTime epoch)
```

Comment:
You're using a format like YYYYMM now for table name suffices, right? So can't you do away with the epoch number conversions now?